File: .config/hiss/testdata/HISS-01/c/gap/direct-recursion.c
// SPDX-FileCopyrightText: 2026 lusoris <lusoris@pm.me>
//
// SPDX-License-Identifier: EUPL-1.2

/* The native scanner is a line matcher: it decides goto and nothing else,
   so this cycle in the call graph passes unreported. */
int factorial(int n)
{
	if (n <= 1)
		return 1;
	return n * factorial(n - 1);
}

File: .config/hiss/testdata/HISS-01/go/positive/direct-recursion.go
// SPDX-FileCopyrightText: 2026 lusoris <lusoris@pm.me>
//
// SPDX-License-Identifier: EUPL-1.2

// Package p demonstrates a call graph that is not a DAG.
package p

// Factorial calls itself, so the call graph contains a cycle.
func Factorial(n int) int {
	if n <= 1 {
		return 1
	}
	return n * Factorial(n-1)
}
